docs/permissions.md

```diff
@@ -0,0 +1,142 @@
+
+### API enablement
+
+Ensure that Datastream and Dataflow apis are enabled on your project.
+
+1. [Make sure that billing is enabled for your Google Cloud project](https://cloud.google.com/billing/docs/how-to/verify-billing-enabled#gcloud).
+2. Follow the [Datastream guidelines](https://cloud.google.com/datastream/docs/use-the-datastream-api#enable_the_api) to enable Datasteream api.
+3. Enable the Dataflow api by using:
+
+   ```sh
+   gcloud services enable dataflow.googleapis.com
+   ```
+
+4. Google Cloud Storage apis are generally enabled by [default](https://cloud.google.com/service-usage/docs/enabled-service#default). In they have been disabled, you will need to enable them.
+
+   ```sh
+   gcloud services enable storage.googleapis.com
+   ```
+
+### Configuring connectivity for `spanner-migration-tool`
+
+In order for SMT to read the information schema from the source database, ensure that the machine where you run `spanner-migration-tool` is allowlisted to connect to the source database.
+In generic terms (your specific network settings may differ), do the following:
+
+1. Open your source database machine's network firewall rules.
+2. Create an inbound rule.
+3. Set the source ip address as the ip address of the machine where you run the `spanner-migration-tool`.
+4. Set the protocol to TCP.
+5. Set the port associated with the TCP protocol of your database.
+6. Save the firewall rule, and then exit.
+
+### Configure Datastream connectivity to source database
+
+{: .highlight }
+This is only needed for minimal downtime migrations via Spanner migration tool.
+
+Follow the [Datastream guidelines](https://cloud.google.com/datastream/docs/network-connectivity-options) to allowlist datastream to access the source database.
+
+- [IP allowlist](https://cloud.google.com/datastream/docs/network-connectivity-options#ipallowlists)
+- [Forward SSH Tunneling](https://cloud.google.com/datastream/docs/network-connectivity-options#sshtunnel)
+- [VPC Peering](https://cloud.google.com/datastream/docs/network-connectivity-options#privateconnectivity)
+
+### Configure source database to enable CDC capture via Datastream
+
+{: .highlight }
+This is only needed for minimal downtime migrations via Spanner migration tool.
+
+Even if the source database is reachable via Datastream, certain prerequisites
+need to be performed on the source database before Datastream can streaming
+backfill and CDC events from it. The steps required vary for each database.
+Validate that the following steps have been performed on the source database
+before using SMT.
+
+- [MySQL](https://cloud.google.com/datastream/docs/configure-your-source-mysql-database)
+- [Postgres](https://cloud.google.com/datastream/docs/configure-your-source-postgresql-database)
+- [Oracle](https://cloud.google.com/datastream/docs/configure-your-source-oracle-database)
+
+### Configuring connectivity for Dataflow
+
+{: .highlight }
+This is only required when if you plan to run Dataflow inside a VPC.
```

Comment:
where do we mention setting the subnet, service account etc ?

Reply:
This is the permissions that Dataflow needs to run. Subnet, service account etc are configurations passed to a Dataflow *job*.